packages/server/src/api/controllers/query/index.ts

```diff
@@ -155,51 +231,16 @@ export async function preview(ctx: UserCtx) {
     const nestedSchemaFields: {
       [key: string]: Record<string, string | QuerySchema>
     } = {}
-    const makeQuerySchema = (
-      type: FieldType,
-      name: string,
-      subtype?: string
-    ): QuerySchema => ({
-      type,
-      name,
-      subtype,
-    })
     if (rows?.length > 0) {
       for (let key of [...new Set(keys)] as string[]) {
-        const field = rows[0][key]
-        let type = typeof field,
-          fieldMetadata = makeQuerySchema(FieldType.STRING, key)
-        if (field)
-          switch (type) {
-            case "boolean":
-              fieldMetadata = makeQuerySchema(FieldType.BOOLEAN, key)
-              break
-            case "object":
-              if (field instanceof Date) {
-                fieldMetadata = makeQuerySchema(FieldType.DATETIME, key)
-              } else if (Array.isArray(field)) {
-                if (JsonUtils.hasSchema(field[0])) {
-                  fieldMetadata = makeQuerySchema(
-                    FieldType.JSON,
-                    key,
-                    JsonFieldSubType.ARRAY
-                  )
-                } else {
-                  fieldMetadata = makeQuerySchema(FieldType.ARRAY, key)
-                }
-                nestedSchemaFields[key] = getSchemaFields(
-                  field,
-                  Object.keys(field[0])
-                ).previewSchema
```

Comment:
I don't seem to be able to find where you've moved this to. I was looking for usages of `getSchemaFields` and I only find it here, not in any of the new code. What am I missing?

Reply:
`getSchemaFields` is called on line 270: 

```ts
const { previewSchema, nestedSchemaFields } = getSchemaFields(rows, keys)
```

Originally this code was recursive:
```ts
nestedSchemaFields[key] = getSchemaFields(
                  field,
                  Object.keys(field[0])
                ).previewSchema
```

However because of the change to use `buildNestedSchema` which loops over all the rows to generate the schema, I removed the recursive aspect.